tdrs-backend/tdpservice/parsers/parse.py

```diff
@@ -273,8 +275,10 @@ def parse_datafile_lines(datafile, program_type, section, is_encrypted):
     return errors
 
 
-def manager_parse_line(line, schema_manager, generate_error, is_encrypted=False):
+def manager_parse_line(line, schema_manager, generate_error, datafile, is_encrypted=False):
     """Parse and validate a datafile line using SchemaManager."""
+    if type(schema_manager) is row_schema.SchemaManager:
```

Comment:
Why are we doing this type check? Shouldnt this always be true?

Reply:
for ba files, when schema manager can't be found, then it will be a list, so this is a check to make sure schema_manager is valid